backend/api_handler.py

```diff
@@ -60,27 +66,6 @@ def adapted(obj, info, **kwargs):
 print(f'Lambda Context ' f'Initialization took: {end - start:.3f} sec')
 
 
-def get_cognito_groups(claims):
-    if not claims:
-        raise ValueError(
-            'Received empty claims. ' 'Please verify authorizer configuration',
-            claims,
-        )
-    groups = list()
-    saml_groups = claims.get('custom:saml.groups', '')
-    if len(saml_groups):
-        groups: list = saml_groups.replace('[', '').replace(']', '').replace(', ', ',').split(',')
-    cognito_groups = claims.get('cognito:groups', '')
-    if len(cognito_groups):
-        groups.extend(cognito_groups.split(','))
-    return groups
-
-
-def get_custom_groups(user_id):
-    service_provider = ServiceProviderFactory.get_service_provider_instance()
-    return service_provider.get_groups_for_user(user_id)
-
-
 def handler(event, context):
     """Sample pure Lambda function
```

Comment:
the handler function looks very messy, it is trying to do many things:
- setting the api context
- adding tenant permissions
- checking maintenance
- checking reauth
Can we divide it into smaller functions?

Reply:
Yes. Updated in the next commit

Reply:
Sure. I can do that . I see the value of reusing them. Thanks @petrkalos!